File: SampleApp/Objective-C/SampleApp/MoPub/RewardedVideo/MoPubRewardedVideoViewController.m
//
//  MoPubRewardedVideoViewController.m
//  SampleApp
//
//  Copyright © 2020 fluct, inc. All rights reserved.
//

#import "MoPubRewardedVideoViewController.h"
#import <FluctSDK/FluctSDK.h>
#import <MoPubMediationAdapterFluct/FluctInstanceMediationSettings.h>
#import <MoPubSDK/MoPub.h>

@interface MoPubRewardedVideoViewController () <MPRewardedAdsDelegate>
@property (nonatomic, weak) IBOutlet UIButton *showButton;
@end

// AdUnitIDを適切なものに変えてください
static NSString *const kAdUnitID = @"8d14fd46f8a449f8a5f1de814e4f5fde";

@implementation MoPubRewardedVideoViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    [MPRewardedAds setDelegate:self forAdUnitId:kAdUnitID];
}

- (IBAction)didTouchUpLoadAdWithButton:(UIButton *)button {
    FSSRewardedVideoSetting *setting = [FSSRewardedVideoSetting defaultSetting];
    setting.debugMode = YES;
    setting.activation.unityAdsActivated = NO;

    FluctInstanceMediationSettings *mediationSetting = [FluctInstanceMediationSettings new];
    mediationSetting.setting = setting;
    [MPRewardedAds loadRewardedAdWithAdUnitID:kAdUnitID withMediationSettings:@[ mediationSetting ]];
}

- (IBAction)didTouchUpShowAdWithButton:(UIButton *)button {
    if ([MPRewardedAds hasAdAvailableForAdUnitID:kAdUnitID]) {
        MPReward *reward = [MPRewardedAds selectedRewardForAdUnitID:kAdUnitID];
        [MPRewardedAds presentRewardedAdForAdUnitID:kAdUnitID fromViewController:self withReward:reward];
    }
}

#pragma mark - MPRewardedAdsDelegate

- (void)rewardedAdDidLoadForAdUnitID:(NSString *)adUnitID {
    NSLog(@"%s", __FUNCTION__);
    self.showButton.enabled = YES;
}

- (void)rewardedAdDidFailToLoadForAdUnitID:(NSString *)adUnitID error:(NSError *)error {
    NSLog(@"%s, %@", __FUNCTION__, error);
}

- (void)rewardedAdWillPresentForAdUnitID:(NSString *)adUnitID {
    NSLog(@"%s", __FUNCTION__);
}

- (void)rewardedAdDidPresentForAdUnitID:(NSString *)adUnitID {
    NSLog(@"%s", __FUNCTION__);
}

- (void)rewardedAdDidFailToShowForAdUnitID:(NSString *)adUnitID error:(NSError *)error {
    NSLog(@"%s, %@", __FUNCTION__, error);
}

- (void)rewardedAdShouldRewardForAdUnitID:(NSString *)adUnitID reward:(MPReward *)reward {
    NSLog(@"%s, %@", __FUNCTION__, reward);
}

- (void)rewardedAdDidExpireForAdUnitID:(NSString *)adUnitID {
    NSLog(@"%s", __FUNCTION__);
}

- (void)rewardedAdDidReceiveTapEventForAdUnitID:(NSString *)adUnitID {
    NSLog(@"%s", __FUNCTION__);
}

- (void)rewardedAdWillLeaveApplicationForAdUnitID:(NSString *)adUnitID {
    NSLog(@"%s", __FUNCTION__);
}

- (void)rewardedAdWillDismissForAdUnitID:(NSString *)adUnitID {
    NSLog(@"%s", __FUNCTION__);
}

- (void)rewardedAdDidDismissForAdUnitID:(NSString *)adUnitID {
    NSLog(@"%s", __FUNCTION__);
    self.showButton.enabled = NO;
}

@end

File: SampleApp/Swift/SampleApp/MoPub/RewardedVideo/MoPubRewardedVideoViewController.swift
//
//  MoPubRewardedVideoViewController.swift
//  SampleApp
//
//  Copyright © 2020 fluct, Inc. All rights reserved.
//

import UIKit
import MoPubSDK
import FluctSDK
import MoPubMediationAdapterFluct

// AdUnitIDを適切なものに変えてください
private let adUnitID: String = "8d14fd46f8a449f8a5f1de814e4f5fde"

class MoPubRewardedVideoViewController: UIViewController {

    @IBOutlet weak var showButton: UIButton!

    override func viewDidLoad() {
        super.viewDidLoad()

        // Do any additional setup after loading the view.
        MPRewardedAds.setDelegate(self, forAdUnitId: adUnitID)
    }

    @IBAction func didTouchUpLoadAd(button: UIButton) {
        let setting = FSSRewardedVideoSetting.default
        setting.isDebugMode = true
        setting.activation.isUnityAdsActivated = false
        let mediationSetting = FluctInstanceMediationSettings()
        mediationSetting.setting = setting
        MPRewardedAds.loadRewardedAd(withAdUnitID: adUnitID, withMediationSettings: [setting])
    }

    @IBAction func didTouchUpShowAd(button: UIButton) {
        guard MPRewardedAds.hasAdAvailable(forAdUnitID: adUnitID) else {
            return
        }

        let reward = MPRewardedAds.selectedReward(forAdUnitID: adUnitID)
        MPRewardedAds.presentRewardedAd(forAdUnitID: adUnitID, from: self, with: reward)
    }

}

// MARK: - MPRewardedVideoDelegate

extension MoPubRewardedVideoViewController: MPRewardedAdsDelegate {

    func rewardedAdDidLoad(forAdUnitID adUnitID: String!) {
        print(#function)
        showButton.isEnabled = true
    }

    func rewardedAdDidFailToLoad(forAdUnitID adUnitID: String!, error: Error!) {
        print(#function, error!)
    }

    func rewardedAdWillPresent(forAdUnitID adUnitID: String!) {
        print(#function)
    }

    func rewardedAdDidPresent(forAdUnitID adUnitID: String!) {
        print(#function)
    }

    func rewardedAdDidFailToShow(forAdUnitID adUnitID: String!, error: Error!) {
        print(#function, error!)
    }

    func rewardedAdShouldReward(forAdUnitID adUnitID: String!, reward: MPReward!) {
        print(#function, reward!)
    }

    func rewardedAdDidExpire(forAdUnitID adUnitID: String!) {
        print(#function)
    }

    func rewardedAdDidReceiveTapEvent(forAdUnitID adUnitID: String!) {
        print(#function)
    }

    func rewardedAdWillLeaveApplication(forAdUnitID adUnitID: String!) {
        print(#function)
    }

    func rewardedAdWillDismiss(forAdUnitID adUnitID: String!) {
        print(#function)
    }

    func rewardedAdDidDismiss(forAdUnitID adUnitID: String!) {
        print(#function)
        showButton.isEnabled = false
    }
}
